### PR TITLE
Cancel reconciliation when chartoperator resource cannot complete

### PR DESCRIPTION
**Pull request by rossf7**
Follow up to https://github.com/giantswarm/app-operator/pull/142

See https://github.com/giantswarm/app-operator/pull/142#discussion_r329430786

### CODE REVIEW

File: service/controller/app/v1/resource/chartoperator/resource.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func (r Resource) installChartOperator(ctx context.Context, cr v1alpha1.App) err
 		if apierrors.IsNotFound(err) {
 			r.logger.LogCtx(ctx, "level", "debug", "message", "can't find chart-operator app CR")
 			r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling the resource")
-			resourcecanceledcontext.SetCanceled(ctx)
+			reconciliationcanceledcontext.SetCanceled(ctx)
```
Comment (rossf7) [inline service/controller/app/v1/resource/chartoperator/resource.go]: Without these we can't install chart-operator so there is no point continuing with other resources.

**Comment by tomahawk28** [comment]
```{"caller":"github.com/giantswarm/app-operator/vendor/github.com/giantswarm/e2e-harness/pkg/release/release.go:404","level":"debug","message":"failed to get release version '0.6.8': retrying in 1m11.521475241s","stack":"waiting for '0.6.8', current '0.6.7': release version not matching error","time":"2019-09-30T08:03:09.851361+00:00"}
--- FAIL: TestAppLifecycleUsingKubeconfig (491.50s)``` LGTM but no idea why update failed.

**Comment by rossf7** [comment]
> --- FAIL: TestAppLifecycleUsingKubeconfig (491.50s)

Worked on the retry looks like its flakey 😞 I'll try to track down the cause.

**Comment by rossf7** [comment]
Stale. Closing in favor of #155.

**Pull request closed without merging**